static/app/components/workflowEngine/gridCell/actionCell.tsx

```diff
@@ -4,37 +4,35 @@ import {Flex} from 'sentry/components/container/flex';
 import {Tooltip} from 'sentry/components/core/tooltip';
 import {IconCircledNumber} from 'sentry/components/iconCircledNumber';
 import {EmptyCell} from 'sentry/components/workflowEngine/gridCell/emptyCell';
-import {IconMail} from 'sentry/icons';
-import {t} from 'sentry/locale';
-import {PluginIcon} from 'sentry/plugins/components/pluginIcon';
+import {ActionMetadata} from 'sentry/components/workflowEngine/ui/actionMetadata';
 import {space} from 'sentry/styles/space';
-
-const ActionMetadata = {
-  slack: {name: t('Slack'), icon: <PluginIcon pluginId="slack" size={20} />},
-  discord: {name: t('Discord'), icon: <PluginIcon pluginId={'discord'} size={20} />},
-  email: {name: t('Email'), icon: <IconMail size="md" />},
-};
-
-export type Action = keyof typeof ActionMetadata;
+import type {ActionType} from 'sentry/types/workflowEngine/actions';
 
 type ActionCellProps = {
-  actions: Action[];
+  actions: ActionType[];
   disabled?: boolean;
 };
 
 export function ActionCell({actions, disabled}: ActionCellProps) {
   if (!actions || actions.length === 0) {
     return <EmptyCell />;
   }
+
   if (actions.length === 1 && actions[0]) {
+    const {name, icon} = ActionMetadata[actions[0]]!;
```

Comment:
We should probably not use `!` here, but actually handle the exception

Reply:
added handling for the exception here in https://github.com/getsentry/sentry/pull/90351/commits/a4eb953e49e5f13f7cab2c500a93c4f6d9a3e98f,  if there's no metadata we'll just return an `<EmptyCell />`. maybe in the future we'll want to return an error icon or something for misconfigured actions?